scripts/submit_sim_imcalc.py

```diff
@@ -89,237 +36,146 @@ def submit_im_calc_slurm(
-        SlBodyOptConsts.sim_name.value: sim_name,
-        SlBodyOptConsts.fault_name.value: fault_name,
-        SlBodyOptConsts.n_procs.value: options_dict[SlBodyOptConsts.n_procs.value],
-        SlBodyOptConsts.extended.value: "-e"
-        if options_dict[SlBodyOptConsts.extended.value]
-        else "",
-        SlBodyOptConsts.simple_out.value: "-s"
-        if options_dict[SlBodyOptConsts.simple_out.value]
-        else "",
-        SlBodyOptConsts.advanced_IM.value: "-a {}".format(
-            " ".join(options_dict[SlBodyOptConsts.advanced_IM.value])
+        command_options[
+            const.SlBodyOptConsts.advanced_IM.value
+        ] = f"-a {body_options['models']}"
+
+        header_options[const.SlHdrOptConsts.additional.value].append(
+            "#SBATCH --nodes=1"
         )
-        if options_dict[SlBodyOptConsts.advanced_IM.value]
-        else "",
-        "pSA_periods": f"-p {' '.join(str(p) for p in params['pSA_periods'])}"
-        if "pSA_periods" in params
-        else "",
-    }
+        header_options[const.SlHdrOptConsts.n_tasks.value] = body_options["np"] = 18
+
+        # Time for one station to run in hours
+        # This should be a machine property. Or take the largest across all machines used
+        time_for_one_station = 12 / 60
+        est_run_time = np.ceil(station_count / 40) * 2 * time_for_one_station
 
-    # determine script template based on advanced_IM or not
-    if options_dict[SlBodyOptConsts.advanced_IM.value]:
-        sl_template = "adv_im_calc.sl.template"
-        script_prefix = "adv_im_calc"
     else:
+        proc_type = const.ProcessType.IM_calculation
         sl_template = "sim_im_calc.sl.template"
         script_prefix = "sim_im_calc"
 
-    body_template_params = (
-        sl_template,
-        {
-            "component": options_dict[SlBodyOptConsts.component.value],
-            "sim_name": sim_name,
-            "fault_name": fault_name,
-            "np": options_dict[SlBodyOptConsts.n_procs.value],
-            "output_csv": sim_struct.get_IM_csv(sim_dir),
-            "output_info": sim_struct.get_IM_info(sim_dir),
-            "models": " ".join(options_dict[SlBodyOptConsts.advanced_IM.value])
-            if options_dict[SlBodyOptConsts.advanced_IM.value]
-            else None,
-        },
-    )
-    script_file_path = write_sl_script(
-        options_dict["write_directory"],
-        sim_dir,
-        proc_type,
-        script_prefix,
-        header_dict,
-        body_template_params,
-        command_template_parameters,
-    )
+        if simple_out:
+            command_options[const.SlBodyOptConsts.simple_out.value] = "-s"
 
-    submit_yes = (
-        True if options_dict["auto"] else confirm("Also submit the job for you?")
-    )
+        if params["ims"][const.RootParams.extended_period.name]:
+            command_options[const.SlBodyOptConsts.extended.value] = "-e"
+            period_count = 100
+        else:
+            period_count = 15
```

Comment:
31 / 131 But yeah not generally a fan of this. Unsure how to do it better though.

Reply:
Oh, pSA_periods is a variable I can take the length of. Getting the EE period count as a variable would probably require importing from IM_calc though

Reply:
That sounds promising. Don't you also need to import from IM_calc at some point anyway? Or is it separated well enough?

Reply:
Tada, just needed an import from qcore.constants